atlasdb-config/src/main/java/com/palantir/atlasdb/factory/InitialisingTransactionManager.java

```diff
@@ -0,0 +1,630 @@
+import com.palantir.atlasdb.sweep.SweepBatchConfig;
+import com.palantir.atlasdb.sweep.SweepMetrics;
+import com.palantir.atlasdb.sweep.SweepTaskRunner;
+import com.palantir.atlasdb.sweep.SweeperServiceImpl;
+import com.palantir.atlasdb.table.description.Schema;
+import com.palantir.atlasdb.table.description.Schemas;
+import com.palantir.atlasdb.transaction.api.AtlasDbConstraintCheckingMode;
+import com.palantir.atlasdb.transaction.api.KeyValueServiceStatus;
+import com.palantir.atlasdb.transaction.api.LockAcquisitionException;
+import com.palantir.atlasdb.transaction.api.LockAwareTransactionTask;
+import com.palantir.atlasdb.transaction.api.TransactionFailedRetriableException;
+import com.palantir.atlasdb.transaction.api.TransactionTask;
+import com.palantir.atlasdb.transaction.impl.ConflictDetectionManager;
+import com.palantir.atlasdb.transaction.impl.ConflictDetectionManagers;
+import com.palantir.atlasdb.transaction.impl.RawTransaction;
+import com.palantir.atlasdb.transaction.impl.SerializableTransactionManager;
+import com.palantir.atlasdb.transaction.impl.SerializableTransactionManagerImpl;
+import com.palantir.atlasdb.transaction.impl.SweepStrategyManager;
+import com.palantir.atlasdb.transaction.impl.SweepStrategyManagers;
+import com.palantir.atlasdb.transaction.impl.TransactionTables;
+import com.palantir.atlasdb.transaction.service.TransactionService;
+import com.palantir.atlasdb.transaction.service.TransactionServices;
+import com.palantir.atlasdb.util.AtlasDbMetrics;
+import com.palantir.atlasdb.util.JavaSuppliers;
+import com.palantir.leader.LeaderElectionService;
+import com.palantir.leader.PingableLeader;
+import com.palantir.leader.proxy.AwaitingLeadershipProxy;
+import com.palantir.lock.HeldLocksToken;
+import com.palantir.lock.InitialisingRemoteLockService;
+import com.palantir.lock.LockClient;
+import com.palantir.lock.LockRefreshToken;
+import com.palantir.lock.LockRequest;
+import com.palantir.lock.LockServerOptions;
+import com.palantir.lock.RemoteLockService;
+import com.palantir.lock.SimpleTimeDuration;
+import com.palantir.lock.client.LockRefreshingRemoteLockService;
+import com.palantir.lock.impl.LockServiceImpl;
+import com.palantir.timestamp.InitialisingTimestampService;
+import com.palantir.timestamp.TimestampService;
+import com.palantir.timestamp.TimestampStoreInvalidator;
+
+public class InitialisingTransactionManager extends ForwardingObject implements SerializableTransactionManager {
+
+    public static final Logger log = LoggerFactory.getLogger(InitialisingTransactionManager.class);
+    private static final int LOGGING_INTERVAL = 60;
+    private static final LockClient LOCK_CLIENT = LockClient.of("atlas instance");
+
+    private volatile SerializableTransactionManagerImpl delegate = null;
+    private Leaders.LocalPaxosServices localPaxosServices = null;
+    private final InitialisingRemoteLockService remoteLockService = InitialisingRemoteLockService.create();
```

Comment:
Why do we have a `InitialisingRemoteLockService`? LockServiceImpl.create doesnt need the KVS at all.